src/engine/cachingreader/cachingreaderchunk.h

```diff
@@ -15,68 +15,76 @@
 // and the worker.
 class CachingReaderChunk {
 public:
-    static const mixxx::audio::ChannelCount kChannels;
-    static const SINT kFrames;
-    static const SINT kSamples;
-
-    // Converts frames to samples
-    static SINT frames2samples(SINT frames) {
-        return frames * kChannels;
-    }
-    static double dFrames2samples(SINT frames) {
-        return static_cast<double>(frames) * kChannels;
-    }
+  // One chunk should contain 1/2 - 1/4th of a second of audio.
+  // 8192 frames contain about 170 ms of audio at 48 kHz, which
+  // is well above (hopefully) the latencies people are seeing.
+  // At 10 ms latency one chunk is enough for 17 callbacks.
+  // Additionally the chunk size should be a power of 2 for
+  // easier memory alignment.
+  // TODO(XXX): The optimum value of the "constant" kFrames depends
+  // on the properties of the AudioSource as the remarks above suggest!
+  static const constexpr mixxx::audio::ChannelCount kChannels = mixxx::kEngineChannelCount;
+  static const constexpr SINT kFrames = 8192; // ~ 170 ms at 48 kHz
+  static const constexpr SINT kSamples = kFrames * kChannels;
+
+  // Converts frames to samples
+  [[nodiscard]] static constexpr SINT frames2samples(SINT frames) noexcept {
+      return frames * kChannels;
+  }
+  [[nodiscard]] static constexpr double dFrames2samples(SINT frames) noexcept {
```

Comment:
We have decided not to maintain [[nodiscard]] in our code. 

https://github.com/mixxxdj/mixxx/wiki/Coding-Guidelines#attribute-nodiscard

Reply:
> Use it when not checking the return value is a program error like a memory leak. Don't use it everywhere, like in simple getter functions where the compiler will discard the call if the return value is not used.

How to know if the compiler will discard the call? This feels like a condition that no one can predict, especially when the project is built with many different compilers.

Reply:
Also, the implementation might be inline and simple at day 1, but maybe day 2, it isn't inline anymore, complicated and not optimized out by the compiler when the return value is discarded.

Reply:
I don't mean to start trolling your coding convention, if you insist I'll change it so it is conform to what you want. Though I fail to see a disadvantage to a function that is annotated with `[[nodiscard]]` when the return value shouldn't be discarded or the function shouldn't be called.

Reply:
We have put the rule in our code style after a discussion to avoid discussing it again in reviews.
Your arguments valid to support your opinion. The disadvantage is the effort to maintain `[[nodiscard]]` everywhere to follow a uniform code style.

Reply:
Can be added step by step, it doesn't have to be done all at once.

Anyway, I'm not here to make you change your mind. I've reverted that change.